WcaOnRails/app/jobs/compute_auxiliary_data.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class ComputeAuxiliaryData < TimedApplicationJob
+  queue_as :default
+
+  def perform
+    # Note: During the results posting process some results may be missing their corresponding WCA ID.
+    #       If we detect that (means someone else is posting results at the moment), we defer the computation.
```

Comment:
Yeah, I thought of that. But the case of simultaneous run is probably very rare, so a case it happens right after this `if` statement and before the query for Concise*Results computation is run - is negligible I'd say. I consider this whole thing as not rally scary because if someone starts posting results, even if the job will finish and do the computation a bit wrong, the the person posting results will run the job very soon anyway (that's how I understand this at least).